### PR TITLE
Conversion to D2-only for v5.0.0

### PR DESCRIPTION
**Pull request by mihails-strasuns**
Includes automatic conversion from d1to2fix and removal of `version(D_Version2)` blocks.
More cleanup will follow in separate PRs after - there are plenty of things to improve but I am not sure how many will be finished by the time I leave.

### CODE REVIEW

File: src/ocean/core/SmartEnum.d
Comment (leandro-lucarella-sociomantic) [inline src/ocean/core/SmartEnum.d:636]: This looks unrelated...

**Review by leandro-lucarella-sociomantic — COMMENTED** [review]
I just gave it a quick look.

I'm really scared now that this is really happening...

File: src/ocean/core/SmartEnum.d
Comment (mihails-strasuns) [reply src/ocean/core/SmartEnum.d:636]: It is related to a change earlier in this module - `enum_id` was a constant defined based on D version. Now it can be removed and only possible value used directly.

**Review by matthias-wende-sociomantic — APPROVED** [review]
Approved apart from the "Remvoe" spelling mistake (in one of the commit messages).

Although given that many of the changes have a fairly automatic nature or maybe even because of that, it might be worth waiting for a second review this time.

File: src/ocean/application/components/PidLock.d
Comment (Geod24) [inline src/ocean/application/components/PidLock.d]: Personally I would suggest not adding those in the first place. They are just noise that needs to be removed afterwards, except for a very few of them which can be reviewed case-by-case.

Comment (mihails-strasuns) [reply src/ocean/application/components/PidLock.d]: That was my first approach (with tweaked d1to2fix) but after trying it in practice I got scared of regressions because cases like `return this` can potentially keep compiling but with a changed meaning. It feels safer to cleanup it in chunks that actually can be reasonably reviewed.

**Comment by mihails-strasuns** [comment]
- Fixed commit title typo

**Review by stefan-koch-sociomantic — APPROVED** [review]
this is the type of PR which can only be done by a blindfold merge.
I give my blindfolded approval.

**Comment by mihails-strasuns** [comment]
Will wait for acknowledgement from @gavin-norman-sociomantic and merge after.

**Comment by mihails-strasuns** [comment]
Updated:
- remove lot of redundant `&this` with some sed

**Pull request merged**
